sdk/src/azure/core/az_log.c

```diff
@@ -16,75 +16,51 @@
 
 #ifndef AZ_NO_LOGGING
 
-static az_log_classification const* volatile _az_log_classifications = NULL;
+// Only using volatile here, not for thread safety, but so that the compiler does not optimize what
+// it falsely thinks are stale reads.
 static az_log_message_fn volatile _az_log_message_callback = NULL;
+static az_log_should_write_fn volatile _az_log_should_write_callback = NULL;
```

Comment:
I can't remember if we default to writing all classifications; but, if so, you could initialize this to point to a function (in core) that simply returns true.

Reply:
Yes, we default to write all if the callback is `NULL`.